nexus/db-model/src/queries/resource_usage_update.rs

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Describes the resource usage update CTE
```

Comment:
Each of these `table` definitions corresponds to one "CTE arm" in the CTE we're construction. That is, we're building:

```sql
WITH
parent_org AS ...,
parent_silo AS ...,
parent_fleet AS ...,
all_collections AS ...,
UPDATE omicron.public.resource_usage SET <values> WHERE id IN all_collections RETURNING <ResourceUsage columns>
```

It originated in https://github.com/oxidecomputer/omicron/pull/1733 - it's by far the easiest way to write CTEs, since it allows type-checking of all the intermediate arms. I elaborated on this a bit in https://github.com/oxidecomputer/omicron/pull/1733/files#diff-e8af81b3111d6041924d2b68cedb35416405b8b481ff0e64c71489196cd45095R11-R23 - I don't think we need the full table, but we *do* need Diesel to generate a named object with named (and strongly-typed) columns. As it happens, the `table` macro does all of that for us.

I understand that this is a bit hacky, but are you very strongly opposed to moving forward with this? I can stop adding CTEs if you think this is the wrong progression.

Reply:
That makes total sense!  I missed that these were for CTE arms.  Thanks for clarifying.  (Mind fleshing out the file-level comment in a bit more detail, even if it's just "see the one in resource_allocation.rs"?)

Reply:
Done in acd4d773e283192d4d8e55cb5a0d008a2f655da6